src/display/components/table.rs

```diff
@@ -50,10 +39,40 @@ fn sort_by_bandwidth<'a, T>(
     list
 }
 
+pub enum ColumnCount {
+    Two,
+    Three,
+}
+
+impl ColumnCount {
+    pub fn as_u16(&self) -> u16 {
+        match &self {
+            ColumnCount::Two => 2,
+            ColumnCount::Three => 3,
+        }
+    }
+}
+
+pub struct ColumnData {
+    column_count: ColumnCount,
+    column_widths: Vec<u16>,
+}
+
 pub struct Table<'a> {
     title: &'a str,
     column_names: &'a [&'a str],
     rows: Vec<Vec<String>>,
+    breakpoints: BTreeMap<u16, ColumnData>,
+}
+
+fn truncate_middle(row: &str, max_length: u16) -> String {
+    if row.len() as u16 > max_length {
+        let first_slice = &row[0..(max_length as usize / 2) - 2];
```

Comment:
Not very important as the inputs are literals, but we could enforce that
* max_length / 2 >= 2
* row.len() >= max_length / 2 + 2

To avoid invalid indexes

Reply:
It's a good point... I had this in mind, but couldn't think about what I'd like to happen if that's the case. Do you have any ideas?

Reply:
hah, good question. This should only happen if the max_length is so small that the `[..]` itself does not fit and/or the hostname is also too small. Not sure, but maybe just as many dots as it fits, or just a dash?

Reply:
IMO this can only be a programmer error, since these values are hard-coded. Maybe just an `expect` so at least at runtime they know what went wrong?